File: app/src/main/java/com/example/mysympleapplication/hw9/PostuplenieDao.java
package com.example.mysympleapplication.hw9;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.mysympleapplication.hw9.model.Postuplenie;
import com.example.mysympleapplication.hw9.model.SumPostValue;

@Dao
public interface PostuplenieDao {
  @Insert
  void insert(Postuplenie postuplenie);

  @Update
  void update(Postuplenie postuplenie);


    @Query("SELECT SUM(value) as value_post FROM postuplenie GROUP BY strftime(\"%m-%Y\", date) ORDER BY date DESC")      // сумма поступлений за месяц
   SumPostValue getMonthPostuplenie();
}

File: app/src/main/java/com/example/mysympleapplication/hw9/newDesign/data/db/dao/PostuplenieDao.kt
package com.example.mysympleapplication.hw9.newDesign.data.db.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.example.mysympleapplication.hw9.model.SumPostValue
import com.example.mysympleapplication.hw9.newDesign.data.entity_model.PostuplenieEntity

@Dao
interface PostuplenieDao {
    @Insert
    fun insert(postuplenie: PostuplenieEntity)

    @Update
    fun update(postuplenie: PostuplenieEntity)


    @Query("SELECT SUM(value) as value_post FROM postuplenie GROUP BY strftime(\"%m-%Y\", date) ORDER BY date DESC")
    fun getMonthPostuplenie(): SumPostValue?   // сумма поступлений за месяц
}